src/domain/core/src/services/query_service.rs

```diff
@@ -51,16 +52,25 @@ pub trait QueryService: Send + Sync {
         options: QueryOptions,
     ) -> Result<DataFrame, QueryError>;
 
-    /// Returns the schema of the given dataset, if it is already defined by
-    /// this moment, None otherwise
-    async fn get_schema(&self, dataset_ref: &DatasetRef) -> Result<Option<Type>, QueryError>;
+    /// Returns a reference-counted arrow schema of the given dataset, if it is
+    /// already defined by this moment, None otherwise
+    async fn get_schema(
+        &self,
+        dataset_ref: &DatasetRef,
+    ) -> Result<Option<arrow::datatypes::SchemaRef>, QueryError>;
 
     // TODO: Introduce additional options that could be used to narrow down the
     // number of files we collect to construct the dataframe.
     //
     /// Returns a [DataFrame] representing the contents of an entire dataset
     async fn get_data(&self, dataset_ref: &DatasetRef) -> Result<DataFrame, QueryError>;
 
+    /// Returns parquet schema of the given dataset, if it is already defined by
+    /// this moment, None otherwise
+    async fn get_schema_parquet(
```

Comment:
Let's keep the schema methods side by side so interface is easier to read

Reply:
what do you mean here?